### PR TITLE
Generalize noise for kriging

### PR DESCRIPTION
**Pull request by Paul-Saves**
Makes the code more clear. Every noise is treated similarly. 

sm = KRG(hyper_opt="Cobyla",eval_noise=True)
image

### CODE REVIEW

File: smt/surrogate_models/krg_based.py [hunk 1]
```diff
@@ -116,12 +124,16 @@ def _new_train(self):
...
             if self.options["eval_noise"]:
                 self.optimal_theta = self.optimal_theta[:-1]
-        elif self.name in ["MGP"]:
-            self._specific_train()
+    
+
+
+
 
         # if self.name != "MGP":
```
Comment (relf) [inline smt/surrogate_models/krg_based.py:138]: to be removed

File: smt/surrogate_models/krg_based.py [hunk 2]
```diff
@@ -116,12 +124,16 @@ def _new_train(self):
...
+        ) = self._optimize_hyperparam(D)         
+        if self.name in ["MGP"]:
+            self._specific_train()
+        else:
             if self.options["eval_noise"]:
                 self.optimal_theta = self.optimal_theta[:-1]
-        elif self.name in ["MGP"]:
-            self._specific_train()
+    
+
```
Comment (relf) [inline smt/surrogate_models/krg_based.py:134]: too many blank lines?

**Review by relf — CHANGES_REQUESTED** [review]
You should pass black on modified files, specially krg_based.py

File: smt/surrogate_models/krg_based.py (hunk 2 above)
Comment (Paul-Saves) [reply smt/surrogate_models/krg_based.py:134]: Yes, sorry

**Pull request merged**